tests/unittests/benchmark/test_benchmark_client.py

```diff
@@ -223,3 +223,96 @@ def test_create_with_invalid_targets(self, benchmark_algorithms):
             assert "type object '{}' has no attribute ".format("DummyAssess") in str(
                 exc.value
             )
+
+    def test_create_with_not_loaded_targets(self, benchmark_algorithms):
+        """Test creation with assessment or task does not exist or not loaded"""
+        cfg_assess = {
+            "name": "bm00002",
+            "algorithms": benchmark_algorithms,
+            "targets": [
+                {
+                    "assess": {
+                        "idontexist": {"task_num": 2},
+                        "AverageRank": {"task_num": 2},
+                    },
+                    "task": {
+                        "RosenBrock": {"dim": 3, "max_trials": 25},
+                        "CarromTable": {"max_trials": 20},
+                    },
+                }
+            ],
+        }
+        with OrionState(benchmarks=cfg_assess):
+            with pytest.raises(NotImplementedError) as exc:
+                get_or_create_benchmark("bm00002")
+            assert "Could not find implementation of BaseAssess" in str(exc.value)
+
+        cfg_task = {
+            "name": "bm00002",
+            "algorithms": benchmark_algorithms,
+            "targets": [
+                {
+                    "assess": {
+                        "AverageResult": {"task_num": 2},
+                        "AverageRank": {"task_num": 2},
+                    },
+                    "task": {
+                        "RosenBrock": {"dim": 3, "max_trials": 25},
+                        "idontexist": {"max_trials": 20},
+                    },
+                }
+            ],
+        }
+
+        with OrionState(benchmarks=cfg_task):
+            with pytest.raises(NotImplementedError) as exc:
+                get_or_create_benchmark("bm00002")
+            assert "Could not find implementation of BaseTask" in str(exc.value)
+
+    def test_create_with_not_exist_targets_parameters(self, benchmark_algorithms):
+        """Test creation with not existing assessment parameters"""
+        cfg = {
```

Comment:
To avoid duplications I would define these configs at the top of the module and modify the small parts as necessary in each tests. To avoid side-effects you can either define them as pytest.fixture or do a deep.copy at the beginning of the test.